File: dcp-solutions/2024-10-02/solution.php
```php
<?php
class Solution {
    function shortestPathBinaryMatrix($matrix) {
        $rows = count($matrix);
        $cols = count($matrix[0]);
        if ($rows === 0 || $cols === 0) return null;
        
        $queue = [[$rows - 1, 0] => ['steps' => 0]];
        while (!empty($queue)) {
            list($r, $c, $path) = array_shift($queue);
            if ($matrix[$r][$c]) continue;
            if (($r === 0 && $c === 0) || count($path) > 0) return $path['steps'];
            $matrix[$r][$c] = true;
            $directions = [[0,1], [0,-1], [1,0], [-1,0]];
            foreach ($directions as $dr) {
                $newR = $r + $dr[0];
                $newC = $c + $dr[1];
                if ($newR >= 0 && $newR < $rows && $newC >= 0 && $newC < $cols 
                    && !$matrix[$newR][$newC]) {
                    $queue[] = [$newR,$newC, ['steps' => count($path) + 1, 'path' => array_merge($path, [$dr])]];
                }
            }
        }
        return null;
    }
}
?>
```

File: dcp-solutions/2024-10-02/solution.py
```python
from collections import deque

def shortestPath(grid, start, end):
    rows, cols = len(grid), len(grid[0])
    directions = [(0, 1), (0, -1), (1, 0), (-1, 0)]
    queue = deque([(start, 0)])
    visited = set([start])

    while queue:
        (x, y), steps = queue.popleft()
        if (x, y) == end:
            return steps
        for dx, dy in directions:
            nx, ny = x + dx, y + dy
            if (0 <= nx < rows and 0 <= ny < cols 
                and grid[nx][ny] == False 
                and (nx, ny) not in visited):
                queue.append(((nx, ny), steps + 1))
                visited.add((nx, ny))

    return None
```